locator/token_metadata.hh

```diff
@@ -133,6 +133,27 @@ private:
     bool _sort_by_proximity = true;
 };
 
+struct host_id_or_endpoint {
```

Comment:
This should maybe be in api/, if the only user of it is there.

Reply:
storage_service needs it to.
I think that since the host id and endpoint are location information this structure should be in the locator namespace. maybe in a separate header file, but we can extract it later to.

In the future, when we use either the endpoint or the host_id to identify nodes, we'll need a registry and a `class node` abstraction.  The mapping registry today is effectively held in the token metadata.

Reply:
Maybe we should extract stuff to locator/topology.hh as a follow up.

Reply:
If it has more users, I'm fine with it staying here.